modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/ModuleOptionsComponent.java

```diff
@@ -572,34 +572,45 @@ public void updateResourceConfiguration(ConfigurationUpdateReport report) {
 
                 //locate specific node and populate the config
                 String loginModuleIdentifier = attribute + ":";
+               
                 int index = path.indexOf(loginModuleIdentifier);
                 String loginModuleIndex = path.substring(index + loginModuleIdentifier.length());
                 String[] split = loginModuleIndex.split(",");
-                int lmi = Integer.valueOf(split[0]);//Ex 0,1,30
-                Value valueObject = currentAttributeState.get(lmi);
-
-                //retrieve current config changes    
-                Configuration conf = report.getConfiguration();
-                //list current conf stated being defined.
-                String mapKey = (String) conf.getMap().keySet().toArray()[0];
-                PropertyMap mapType = (PropertyMap) conf.getMap().get(mapKey);
-                //insert update logic.
-                //set this new state passed in as state of the attribute.
-                LinkedHashMap<String, Object> currentOptions = new LinkedHashMap<String, Object>();
-                for (String propertyKey : mapType.getMap().keySet()) {
-                    currentOptions.put(propertyKey, ((PropertySimple) mapType.get(propertyKey)).getStringValue());
+                
+                Value valueObject=null;
+                int lmi=-1;
+                
+                try{
+                    lmi = Integer.valueOf(split[0]);//Ex 0,1,30
+                    valueObject = currentAttributeState.get(lmi);
+                }catch(NumberFormatException e) {
+                        // Ignore it, the value just wasn't available
                 }
+                
+                if(valueObject!=null){
+                    //retrieve current config changes    
+                    Configuration conf = report.getConfiguration();
+                    //list current conf stated being defined.
+                    String mapKey = (String) conf.getMap().keySet().toArray()[0];
+                    PropertyMap mapType = (PropertyMap) conf.getMap().get(mapKey);
+                    //insert update logic.
+                    //set this new state passed in as state of the attribute.
+                    LinkedHashMap<String, Object> currentOptions = new LinkedHashMap<String, Object>();
+                    for (String propertyKey : mapType.getMap().keySet()) {
+                        currentOptions.put(propertyKey, ((PropertySimple) mapType.get(propertyKey)).getStringValue());
+                    }
 
-                //make these settings the new state for this part of the attribute.
-                valueObject.setOptions(currentOptions);
-                currentAttributeState.set(lmi, valueObject);
+                    //make these settings the new state for this part of the attribute.
+                    valueObject.setOptions(currentOptions);
+                    currentAttributeState.set(lmi, valueObject);
```

Comment:
if the parse of the number fails both `lmi` and `valueObject` will be null and the code inside the `if (valueObject==null) ` won't be executed, but yes I think it's better to put the return, essentially because if I can't parse the number I can't do anything else on that method, so I changed it.